src/otx/v2/adapters/torch/mmengine/utils/runner_config.py

```diff
@@ -4,146 +4,93 @@
-    """Update the training configuration for a PyTorch model training process and mmengine.
-
-    Args:
-        func_args (dict): A dictionary of function arguments.
-        config (Config): A configuration object.
-        precision (str | None): The precision of the model weights (e.g. "float16", "float32").
-        **kwargs: Additional keyword arguments.
-
-    Returns:
-        tuple: A tuple containing the updated configuration object and keyword arguments.
-
-    Raises:
-        ValueError: If both `max_epochs` and `max_iters` are set in the configuration.
 
+    Examples:
+    >>> get_value_from_config(
+        arg_key="max_epochs",
+        positional_args={"max_epochs": 10},
+        config=Config({"max_epochs": 20}),
+    )
+    10
+    >>> get_value_from_config(
+        arg_key="max_epochs",
+        positional_args={},
+        config=Config({"max_epochs": 20}),
+    )
+    20
     """
-    if kwargs.get("train_dataloader", None) is not None:
-        max_iters = get_value_from_config("max_iters", func_args, config=config)
-        max_epochs = get_value_from_config("max_epochs", func_args, config=config)
-        val_interval = get_value_from_config("val_interval", func_args, config=config, default=1)
-        if max_iters is not None and max_epochs is not None:
-            msg = "Only one of `max_epochs` or `max_iters` can be set."
-            raise ValueError(msg)
-        if "train_cfg" not in kwargs or kwargs["train_cfg"] is None:
-            kwargs["train_cfg"] = {"val_interval": val_interval, "by_epoch": True}
-        if max_epochs is not None:
-            kwargs["train_cfg"]["by_epoch"] = True
-            kwargs["train_cfg"]["max_epochs"] = max_epochs
-        elif max_iters is not None:
-            kwargs["train_cfg"]["by_epoch"] = False
-            kwargs["train_cfg"]["max_iters"] = max_iters
-        # Update Optimizer
-        if "optim_wrapper" not in kwargs or kwargs["optim_wrapper"] is None:
-            optimizer = get_value_from_config("optimizer", func_args, config=config)
-            if optimizer is None:
-                optimizer = {"type": "SGD", "lr": 0.01, "momentum": 0.9, "weight_decay": 0.0005}
-            if get_device() not in ("cuda", "gpu", "npu", "mlu"):
-                kwargs["optim_wrapper"] = {"type": "OptimWrapper", "optimizer": optimizer}
-            else:
-                kwargs["optim_wrapper"] = {"type": "AmpOptimWrapper", "dtype": precision, "optimizer": optimizer}
-    elif isinstance(config.get("train_dataloader", None), dict):
-        config["train_dataloader"] = None
-        config["train_cfg"] = None
-        config["optim_wrapper"] = None
-    return config, kwargs
+    # Priority 1: Positional Args
+    result = positional_args.get(arg_key, None)
+    # Priority 2: Input Config Value
+    return config.get(arg_key, None) if result is None else result
 
 
-def update_val_test_config(
-    func_args: dict,
+def update_train_config(
```

Comment:
can I ask why this is declared here with global method, while `_update_config` and `_update_eval_config` are defined at framework engine?

Reply:
Yes, now that I think about it, your comment makes sense. I've modified this to the `MMXEngine._update_train_config` function. Thanks for the nice comment. :)

Reply:
thank you!